src/transformers/models/prism/tokenization_prism.py

```diff
@@ -0,0 +1,426 @@
+
+        self.encoder_size = len(self.encoder)
+
+        self.lang_token_to_id = {
+            self.get_lang_token(lang_code): self.encoder[self.get_lang_token(lang_code)]
+            for lang_code in fairseq_language_code
+            if self.get_lang_token(lang_code) in self.encoder
+        }
+        self.lang_code_to_id = {
+            lang_code: self.encoder[self.get_lang_token(lang_code)]
+            for lang_code in fairseq_language_code
+            if self.get_lang_token(lang_code) in self.encoder
+        }
+        self.id_to_lang_token = {v: k for k, v in self.lang_token_to_id.items()}
+
+        self._src_lang = src_lang if src_lang is not None else "en"
+        self.tgt_lang = tgt_lang
+        self.cur_lang_id = self.get_lang_id(self._src_lang)
+
+        self.num_madeup_words = num_madeup_words
+
+        super().__init__(
+            src_lang=src_lang,
+            tgt_lang=tgt_lang,
+            bos_token=bos_token,
+            eos_token=eos_token,
+            sep_token=sep_token,
+            unk_token=unk_token,
+            pad_token=pad_token,
+            language_codes=language_codes,
+            sp_model_kwargs=self.sp_model_kwargs,
+            additional_special_tokens=additional_special_tokens,
+            num_madeup_words=num_madeup_words,
+            **kwargs,
+        )
+        
+        self.special_tokens_map['additional_special_tokens'] = self.additional_special_tokens
+        self.set_src_lang_special_tokens(self._src_lang)
+
+    @property
+    def vocab_size(self) -> int:
+        return len(self.encoder)
+
+    def get_vocab(self) -> Dict:
+        vocab = {self.convert_ids_to_tokens(i): i for i in range(self.vocab_size)}
+        vocab.update(self.added_tokens_encoder)
+        return vocab
+
+    @property
+    def src_lang(self) -> str:
+        return self._src_lang
+
+    @src_lang.setter
+    def src_lang(self, new_src_lang: str) -> None:
+        self._src_lang = new_src_lang
+        self.set_src_lang_special_tokens(self._src_lang)
+
+    def _tokenize(self, text: str) -> List[str]:
+        return self.sp_model.encode(text, out_type=str)
+
+    def _convert_token_to_id(self, token):
+        if token in self.lang_token_to_id:
+            return self.lang_token_to_id[token]
+        return self.encoder.get(token, self.encoder[self.unk_token])
+
+    def _convert_id_to_token(self, index: int) -> str:
+        """Converts an index (integer) in a token (str) using the decoder."""
+        if index in self.id_to_lang_token:
+            return self.id_to_lang_token[index]
+        return self.decoder.get(index, self.unk_token)
+
+    def convert_tokens_to_string(self, tokens):
+        """Converts a sequence of tokens (string) in a single string."""
+        current_sub_tokens = []
+        out_string = ""
+        for token in tokens:
+            # Skip language tokens during decoding
+            if token in self.lang_code_to_token.values():
+                continue
```

Comment:
I think that if the language tokens are correctly designated as special tokens, this "hack" will become unnecessary.

Reply:
Fixed ✅